File: Ruby/Average Word Length.rb
phrase= gets().chomp()
words=phrase.split(" ")
sum=0
words.each do |s|
    sum += s.length
end
cnt=phrase.count(" ")
avg=sum.to_f/(cnt+1)
if (avg >=3.5)
    puts avg.ceil
else
    puts avg.floor
end

File: Python/Average Word Length.py
import math
from string import punctuation
str=input("Sentence: ").strip()
word0=str.split()
words=[]
for w in word0:
    wd=''
    for ch in w:
        if ch not in punctuation:
           wd += ch
    words.append(wd)
lens= 0
for w in words:
    lens += len(w)
print(math.ceil(lens/len(words)))